trades.py

```diff
@@ -101,19 +140,33 @@ def _wei_to_dai(wei: int) -> str:
     return f"{formatted_dai} DAI"
 
 
-def parse_response(res: requests.Response) -> str:  # pylint: disable=too-many-locals
+def _is_redeemed(user_json: dict[str, Any], condition_id: str) -> bool:
+    user_positions = user_json["data"]["user"]["userPositions"]
+
+    for position in user_positions:
+        position_condition_ids = position["position"]["conditionIds"]
+        balance = int(position["balance"])
+
+        if condition_id in position_condition_ids and balance == 0:
```

Comment:
`balance == 0` won't match the trader's behaviour in case the remaining balance [is dust](https://github.com/valory-xyz/trader/blob/7500873cb0dd66a6da33209a7556e56aecdf5038/packages/valory/skills/decision_maker_abci/behaviours/reedem.py#L312).

Reply:
if balance > 0 --> It has not been redeemed (regardless it's dust or not)
if balance == 0 --> It has been redeemed